### PR TITLE
ci: Modernize CMake usage in CI scripts

### PR DESCRIPTION
**Pull request by juan-lunarg**
- Use Ninja for all platforms (perf/simplicity)
- Specify source/binary dir explitily (-S / -B)

### CODE REVIEW

File: .github/workflows/vvl.yml [hunk 1]
```diff
@@ -196,6 +197,9 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: '3.8'
+      - uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: ${{ matrix.arch }}
```
Comment (juan-lunarg) [inline .github/workflows/vvl.yml:202]: Sets up developer command prompt:
https://github.com/ilammy/msvc-dev-cmd

File: scripts/common_ci.py [hunk 2]
```diff
@@ -234,11 +228,6 @@ def GetArgParser():
         choices=CONFIGURATIONS, default=DEFAULT_CONFIGURATION,
         help='Build target configuration. Can be one of: {0}'.format(
             ', '.join(CONFIGURATIONS)))
-    parser.add_argument(
```
Comment (spencer-lunarg) [inline scripts/common_ci.py:237]: :100:

File: .github/workflows/vvl.yml [hunk 3]
```diff
@@ -103,13 +103,14 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: '3.8'
+      - uses: lukka/get-cmake@latest
```
Comment (juan-lunarg) [inline .github/workflows/vvl.yml:106]: Needed for Ninja

File: scripts/common_ci.py [hunk 4]
```diff
@@ -27,6 +27,9 @@
     print("This script requires Python 3. Run script with [-h] option for more details.")
     sys_exit(0)
 
+# Use Ninja for all platforms for performance/simplicity
+os.environ['CMAKE_GENERATOR'] = "Ninja"
```
Comment (spencer-lunarg) [inline scripts/common_ci.py:31]: :100: x10

**Review by spencer-lunarg — APPROVED** [review]

File: scripts/common_ci.py (hunk 4 above)
Comment (juan-lunarg) [inline scripts/common_ci.py:31]: Much simpler to only have to handle Ninja.

File: scripts/common_ci.py [hunk 5]
```diff
@@ -74,29 +75,29 @@ def CheckVVLCodegenConsistency(args):
...
-
+def BuildVVL(args, build_tests="OFF"):
     print("Log CMake version")
     cmake_ver_cmd = 'cmake --version'
     RunShellCmd(cmake_ver_cmd)
 
-    make_dirs(VVL_BUILD_DIR)
     print("Run CMake for Validation Layers")
-    cmake_cmd = f'cmake -DUPDATE_DEPS=ON -DCMAKE_BUILD_TYPE={args.configuration} {args.cmake} ..'
+    cmake_cmd = f'cmake -S . -B {VVL_BUILD_DIR} -DUPDATE_DEPS=ON -DCMAKE_BUILD_TYPE={args.configuration}'
```
Comment (juan-lunarg) [inline scripts/common_ci.py:84]: CMake will create the build dir if it doesn't exist.

Which removes the need for make_dirs in this scenario.

File: scripts/common_ci.py [hunk 6]
```diff
@@ -74,29 +75,29 @@ def CheckVVLCodegenConsistency(args):
...
-    build_cmd = f'cmake --build . --config {args.configuration}'
-    if not IsWindows(): build_cmd = build_cmd + f' -- -j{os.cpu_count()}'
-    RunShellCmd(build_cmd, VVL_BUILD_DIR)
+    build_cmd = f'cmake --build {VVL_BUILD_DIR}'
+    RunShellCmd(build_cmd)
 
     print("Install Validation Layers")
-    install_cmd = f'cmake --install . --prefix ./install/ --config {args.configuration}'
-    RunShellCmd(install_cmd, VVL_BUILD_DIR)
+    install_cmd = f'cmake --install {VVL_BUILD_DIR} --prefix {VVL_BUILD_DIR}/install/'
```
Comment (juan-lunarg) [inline scripts/common_ci.py:99]: Running this locally results in artifacts that aren't ignored. This makes it easier to run the script locally and blow everything away.

File: scripts/common_ci.py [hunk 7]
```diff
@@ -74,29 +75,29 @@ def CheckVVLCodegenConsistency(args):
...
     print("Run CMake for Validation Layers")
-    cmake_cmd = f'cmake -DUPDATE_DEPS=ON -DCMAKE_BUILD_TYPE={args.configuration} {args.cmake} ..'
+    cmake_cmd = f'cmake -S . -B {VVL_BUILD_DIR} -DUPDATE_DEPS=ON -DCMAKE_BUILD_TYPE={args.configuration}'
     # By default BUILD_WERROR is OFF, CI should always enable it.
-    cmake_cmd = cmake_cmd + ' -DBUILD_WERROR=ON'
-    if IsWindows(): cmake_cmd = cmake_cmd + f' -A {args.arch}'
-    if build_tests: cmake_cmd = cmake_cmd + ' -DBUILD_TESTS=ON'
-    RunShellCmd(cmake_cmd, VVL_BUILD_DIR)
+    cmake_cmd += ' -DBUILD_WERROR=ON'
+    cmake_cmd += f' -DBUILD_TESTS={build_tests}'
```
Comment (juan-lunarg) [inline scripts/common_ci.py:87]: Simpler to just pass `build_tests` directly

**Review by juan-lunarg — COMMENTED** [review]
Overall performance improvement on Windows (3-5 minutes depending on the run)

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**